test/lib/random_schema.cc

```diff
@@ -111,7 +111,11 @@ data_type type_generator::operator()(std::mt19937& engine, is_multi_cell multi_c
     //
     // To cover all this, we simply disallow it altogether when multi_cell is
     // no, which will be the case in all the above cases.
-    while (!multi_cell && type == duration_type) {
+    //
+    // We also disallow boolean type in keys, due to the poor value distribution
+    // it provides. Generating keys which have a boolean in it, are prone to
+    // collision and will result in poor cardinality.
+    while (!multi_cell && (type == duration_type || type == boolean_type)) {
```

Comment:
1. Nitpick (it's pre-existing code) - if you use recursion here, you don't need a while - the random "type" can be bad, but the one returned from the recursive call can't.
2. I don't understand the "!multi_cell" in the condition (again, pre-existing code, but still...)... If a single boolean is bad because it only two options, how is are two or three booleans, with 4 or 8 options, really better?

Reply:
(1) I usually prefer iterations because it leads to easier to understand code, and there is a cap on the stack size. In theory, a long enough sequence of "bad rolls" can lead to stack problem here if recursion was used, although I admit the chance of that is infinitesimally small.
(2) `multi_cell` just designates whether the cell is frozen. Cells are frozen if they are part of a key or are members of a mutation. So this condition actually casts a wider net then necessary, but I decided against making it more precise, to keep the code simple.